Add Maven or Gradle to publish JARs to online repositories

**Describe the solution you'd like**
When a delivery is created the artifacts shall be published on online repositories (Maven Central or GitHub).

**Describe alternatives you've considered**
N/A

**Additional context**
Implemented #25 it is now possible to publish the JARs to central repositories

Building with Maven separated JARs there is this error:
**[ERROR] The projects in the reactor contain a cyclic reference: Edge between 'Vertex{label='com.masesgroup:system.configuration_version_4.0.0.0_culture_neutral_publickeytoken_b03f5f7f11d50a3a:1.4.0.0-SNAPSHOT'}' and 'Vertex{label='com.masesgroup:system_version_4.0.0.0_culture_neutral_publickeytoken_b77a5c561934e089:1.4.0.0-SNAPSHOT'}' introduces to cycle in the graph com.masesgroup:system_version_4.0.0.0_culture_neutral_publickeytoken_b77a5c561934e089:1.4.0.0-SNAPSHOT --> com.masesgroup:system.configuration_version_4.0.0.0_culture_neutral_publickeytoken_b03f5f7f11d50a3a:1.4.0.0-SNAPSHOT --> com.masesgroup:system_version_4.0.0.0_culture_neutral_publickeytoken_b77a5c561934e089:1.4.0.0-SNAPSHOT**.
.NET assemblies cycle-references them and this is not acceptable from Maven. At this moment the compilation and Javadoc is done considering all the Java files reflected, only JARs are built one by one. The same shall be done with Maven with the constraint only one JAR will be built.

--- a/.github/workflows/maven.yaml
+++ b/.github/workflows/maven.yaml
@@ -107,6 +107,10 @@ jobs:
           gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }} # Value of the GPG private key to import
           gpg-passphrase: MAVEN_GPG_PASSPHRASE # env variable for GPG private key passphrase
 
+      - name: Install local file to be used from Maven
+        shell: bash
+        run: mvn install:install-file -Dfile=./bin/net5.0/JCOBridge.jar -DgroupId=JCOBridge -DartifactId=JCOBridge -Dversion=2.3.2 -Dpackaging=jar -DgeneratePom=true
+
       - name: Publish net5.0 to Apache Maven Central
         shell: bash
         run: mvn --file ./netreflected/src/net5.0.xml --no-transfer-progress --batch-mode -Dgpg.passphrase=${{ secrets.MAVEN_GPG_PASSPHRASE }} deploy

--- a/.github/workflows/maven.yaml
+++ b/.github/workflows/maven.yaml
@@ -107,7 +107,7 @@ jobs:
           gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }} # Value of the GPG private key to import
           gpg-passphrase: MAVEN_GPG_PASSPHRASE # env variable for GPG private key passphrase
 
-      - name: Install local file to be used from Maven
+      - name: Install local file to be used within generated POM
         shell: bash
         run: mvn install:install-file -Dfile=./bin/net5.0/JCOBridge.jar -DgroupId=JCOBridge -DartifactId=JCOBridge -Dversion=2.3.2 -Dpackaging=jar -DgeneratePom=true
 

--- a/.github/workflows/maven.yaml
+++ b/.github/workflows/maven.yaml
@@ -56,7 +56,7 @@ jobs:
     if: needs.check.outputs.run_job == 'true' 
     # The type of runner that the job will run on
     runs-on: windows-2019
-  
+
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
       # Runs a set of commands using the runners shell
@@ -126,4 +126,4 @@ jobs:
           MAVEN_USERNAME: ${{ secrets.MAVEN_USERNAME }}
           MAVEN_CENTRAL_TOKEN: ${{ secrets.MAVEN_CENTRAL_TOKEN }}
           MAVEN_GPG_PASSPHRASE: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
-          
+        

--- a/JCOReflector/engine/JCOPOMJCOReflector.template
+++ b/JCOReflector/engine/JCOPOMJCOReflector.template
@@ -64,6 +64,17 @@
     </resources>
     <plugins>
       <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.8.1</version>
+        <configuration>
+          <compilerArgs>
+            <arg>-cp</arg>
+            <arg>../../bin/POM_RUNTIME_PLACEHOLDER/JCOBridge.jar</arg>
+          </compilerArgs>
+        </configuration>
+      </plugin>
+      <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>build-helper-maven-plugin</artifactId>
         <version>3.2.0</version>
@@ -106,8 +117,18 @@ POM_SOURCEDIRECTORIES_PLACEHOLDER
         <artifactId>maven-javadoc-plugin</artifactId>
         <version>2.9.1</version>
         <configuration>
+          <links>
+            <link>https://www.jcobridge.com/api-java</link>
+          </links>
           <quiet>true</quiet>
           <source>8</source>
+          <additionalDependencies>
+            <additionalDependency>
+              <groupId>JCOBridge</groupId>
+              <artifactId>JCOBridge</artifactId>
+              <version>2.3.2</version>
+            </additionalDependency>
+          </additionalDependencies>
         </configuration>
         <executions>
           <execution>
@@ -151,13 +172,4 @@ POM_SOURCEDIRECTORIES_PLACEHOLDER
       </plugin>
     </plugins>
   </build>
-  <dependencies>
-    <dependency>
-      <groupId>JCOBridge</groupId>
-      <artifactId>JCOBridge</artifactId>
-      <version>2.3.2</version>
-      <scope>system</scope>
-      <systemPath>${basedir}\..\..\bin\POM_RUNTIME_PLACEHOLDER\JCOBridge.jar</systemPath>
-    </dependency>
-  </dependencies>
 </project>

--- a/JCOReflector/engine/JCOPOMJCOReflector.template
+++ b/JCOReflector/engine/JCOPOMJCOReflector.template
@@ -71,7 +71,7 @@
         <configuration>
           <compilerArgs>
             <arg>-cp</arg>
-            <arg>../../bin/POM_RUNTIME_PLACEHOLDER/JCOBridge.jar</arg>
+            <arg>${basedir}/../../bin/POM_RUNTIME_PLACEHOLDER/JCOBridge.jar</arg>
           </compilerArgs>
         </configuration>
       </plugin>

--- a/JCOReflector/engine/JCOPOMJCOReflector.template
+++ b/JCOReflector/engine/JCOPOMJCOReflector.template
@@ -58,6 +58,7 @@
         <filtering>false</filtering>
         <includes>
           <include>JCOBridge.zip</include>
+          <include>JCOBridge.jar</include>
         </includes>
         <targetPath>org/mases/jcobridge/netreflection</targetPath>
       </resource>

--- a/netreflected-tests/maven/.idea/.gitignore
+++ b/netreflected-tests/maven/.idea/.gitignore
@@ -1,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml

--- a/netreflected-tests/maven/.idea/.name
+++ b/netreflected-tests/maven/.idea/.name
@@ -1,0 +1,1 @@
+netreflected-tests

--- a/netreflected-tests/maven/.idea/codeStyles/Project.xml
+++ b/netreflected-tests/maven/.idea/codeStyles/Project.xml
@@ -1,0 +1,13 @@
+<component name="ProjectCodeStyleConfiguration">
+  <code_scheme name="Project" version="173">
+    <JetCodeStyleSettings>
+      <option name="CODE_STYLE_DEFAULTS" value="KOTLIN_OFFICIAL" />
+    </JetCodeStyleSettings>
+    <ScalaCodeStyleSettings>
+      <option name="MULTILINE_STRING_CLOSING_QUOTES_ON_NEW_LINE" value="true" />
+    </ScalaCodeStyleSettings>
+    <codeStyleSettings language="kotlin">
+      <option name="CODE_STYLE_DEFAULTS" value="KOTLIN_OFFICIAL" />
+    </codeStyleSettings>
+  </code_scheme>
+</component>

--- a/netreflected-tests/maven/.idea/codeStyles/codeStyleConfig.xml
+++ b/netreflected-tests/maven/.idea/codeStyles/codeStyleConfig.xml
@@ -1,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
+  <state>
+    <option name="PREFERRED_PROJECT_CODE_STYLE" value="Default" />
+  </state>
+</component>

--- a/netreflected-tests/maven/.idea/compiler.xml
+++ b/netreflected-tests/maven/.idea/compiler.xml
@@ -1,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="CompilerConfiguration">
+    <annotationProcessing>
+      <profile name="Maven default annotation processors profile" enabled="true">
+        <sourceOutputDir name="target/generated-sources/annotations" />
+        <sourceTestOutputDir name="target/generated-test-sources/test-annotations" />
+        <outputRelativeToContentRoot value="true" />
+        <module name="netreflected-tests" />
+      </profile>
+    </annotationProcessing>
+  </component>
+</project>

--- a/netreflected-tests/maven/.idea/jarRepositories.xml
+++ b/netreflected-tests/maven/.idea/jarRepositories.xml
@@ -1,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="RemoteRepositoriesConfiguration">
+    <remote-repository>
+      <option name="id" value="central" />
+      <option name="name" value="Central Repository" />
+      <option name="url" value="https://repo.maven.apache.org/maven2" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="central" />
+      <option name="name" value="Maven Central repository" />
+      <option name="url" value="https://repo1.maven.org/maven2" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="jboss.community" />
+      <option name="name" value="JBoss Community repository" />
+      <option name="url" value="https://repository.jboss.org/nexus/content/repositories/public/" />
+    </remote-repository>
+  </component>
+</project>

--- a/netreflected-tests/maven/.idea/misc.xml
+++ b/netreflected-tests/maven/.idea/misc.xml
@@ -1,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ExternalStorageConfigurationManager" enabled="true" />
+  <component name="MavenProjectsManager">
+    <option name="originalFiles">
+      <list>
+        <option value="$PROJECT_DIR$/pom.xml" />
+      </list>
+    </option>
+  </component>
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" project-jdk-name="11" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>

--- a/netreflected-tests/maven/.idea/vcs.xml
+++ b/netreflected-tests/maven/.idea/vcs.xml
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$/../.." vcs="Git" />
+  </component>
+</project>

--- a/netreflected-tests/maven/netreflected-tests.iml
+++ b/netreflected-tests/maven/netreflected-tests.iml
@@ -1,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4" />

--- a/netreflected-tests/maven/pom.xml
+++ b/netreflected-tests/maven/pom.xml
@@ -1,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.masesgroup</groupId>
+    <artifactId>netreflected-tests</artifactId>
+    <version>1.4.0.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+    </properties>
+
+    <build>
+        <sourceDirectory>../java/src</sourceDirectory>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>unpack</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>com.masesgroup</groupId>
+                                    <artifactId>jcoreflector_net5.0</artifactId>
+                                    <outputDirectory>${basedir}</outputDirectory>
+                                    <includes>org/mases/jcobridge/netreflection/JCOBridge.jar</includes>
+                                </artifactItem>
+                            </artifactItems>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.masesgroup</groupId>
+            <artifactId>jcoreflector_net5.0</artifactId>
+            <version>1.4.0.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>JCOBridge</groupId>
+            <artifactId>JCOBridge</artifactId>
+            <version>2.3.2</version>
+            <scope>system</scope>
+            <systemPath>${basedir}/org/mases/jcobridge/netreflection/JCOBridge.jar</systemPath>
+        </dependency>
+    </dependencies>
+</project>

